modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/ddl/DdlCommandHandler.java

```diff
@@ -286,38 +286,42 @@ private CompletableFuture<Boolean> addColumnInternal(String fullName, List<Colum
                 fullName,
-                chng -> chng.changeColumns(cols -> {
-                    ret.set(true); // Reset state if closure have been restarted.
+                chng -> {
+                    chng.changeColumns(cols -> {
+                        ret.set(true); // Reset state if closure have been restarted.
 
-                    Map<String, String> colNamesToOrders = columnOrdersToNames(chng.columns());
+                        Map<String, String> colNamesToOrders = columnOrdersToNames(chng.columns());
 
-                    List<ColumnDefinition> colsDef0;
+                        List<ColumnDefinition> colsDef0;
 
-                    if (ignoreColumnExistance) {
-                        colsDef0 = colsDef.stream().filter(k -> {
-                            if (colNamesToOrders.containsKey(k.name())) {
-                                ret.set(false);
+                        if (ignoreColumnExistance) {
+                            colsDef0 = colsDef.stream().filter(k -> {
+                                if (colNamesToOrders.containsKey(k.name())) {
+                                    ret.set(false);
 
-                                return false;
-                            } else {
-                                return true;
-                            }
-                        }).collect(Collectors.toList());
-                    } else {
-                        colsDef.stream()
-                                .filter(k -> colNamesToOrders.containsKey(k.name()))
-                                .findAny()
-                                .ifPresent(c -> {
-                                    throw new ColumnAlreadyExistsException(c.name());
-                                });
-
-                        colsDef0 = colsDef;
-                    }
+                                    return false;
+                                } else {
+                                    return true;
+                                }
+                            }).collect(Collectors.toList());
+                        } else {
+                            colsDef.stream()
+                                    .filter(k -> colNamesToOrders.containsKey(k.name()))
+                                    .findAny()
+                                    .ifPresent(c -> {
+                                        throw new ColumnAlreadyExistsException(c.name());
+                                    });
+
+                            colsDef0 = colsDef;
+                        }
 
-                    for (ColumnDefinition col : colsDef0) {
-                        cols.create(col.name(), colChg -> convertColumnDefinition(col, colChg));
-                    }
-                })
+                        for (ColumnDefinition col : colsDef0) {
+                            cols.create(col.name(), colChg -> convertColumnDefinition(col, colChg));
+                        }
+                    });
+
+                    return ret.get();
```

Comment:
You can't reuse the same result you're going to return to the user, because the rules are different:

- the result of the whole operation is `true` if _all_ columns have been added
- the result of the change closure is `true` if _at least one_ column has been added (otherwise we won't update the schema version)